### ISSUE TITLE
MatrixClient scrollback does not have server side filtering

### ISSUE THREAD

**Issue by CHANist** [matrix-org/matrix-js-sdk#599]
As seen in client server api r0.3.0 specification, client can query legacy message with filtering, therefore I expect 
MatrixClient.prototype.scrollback = function (room, limit, callback); 
methods to have filtering selection. 

I have also notice there is a method, namely 
MatrixClient.prototype.paginateEventTimeline = function (eventTimeline, opts), 
but have no idea how to use it for filtering. 

Since using such method, will create a filter in timelineSet in the room. 
and when sending new message in the room, 
Room.prototype.addPendingEvent = function (event, txnId) {
   ...
   if (timelineSet.getFilter()) {
        if (this._filter.filterRoomTimeline([event]).length) {
            timelineSet.addEventToTimeline(event, timelineSet.getLiveTimeline(), false);
        }   
   }
}

and this._filter is undefined and throw an exception.

**Comment by ncoevoet** [comment]
i agree, you can't use TimeLineWindow with filter, because there is null pointer there 
https://github.com/matrix-org/matrix-js-sdk/blob/master/src/models/room.js#L661

**Comment by flumpt** [comment]
Changing
`if (this._filter.filterRoomTimeline([event]).length) {`
to
`if (timelineSet.getFilter().filterRoomTimeline([event]).length) {
`
seems to work.

Can somebody confirm, that this is change is correct?

**Comment by flumpt** [comment]
Any news here?

**Comment by CHANist** [comment]
I agree with @flumpt , as this._filter is undefined in room.js, which is probably incorrect in both 
src/models/room.js and lib/models/room.js.

**Comment by flumpt** [comment]
@dbkr Thanks a lot for looking into this and applying the fix!
